qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -99,6 +99,13 @@ def save_page(self):
         """Save the current page."""
         self._widget.triggerPageAction(QWebEnginePage.SavePage)
 
+    def show_source(self, dispatcher, url):
+        # use view-source: scheme to show page source for webengine
+        url = QUrl('view-source:{}'.format(url.toString()))
+        new_tab = dispatcher._tabbed_browser.tabopen(
+            url, background=True, related=True)
+        new_tab.data.viewing_source = True
```

Comment:
If we set a `view-source:` URL for QtWebKit correctly, I actually think we don't need the `viewing_source` flag anymore as we can just check the URL. If that works out, you can remove it entirely.

Reply:
You're right, I was able to get rid of the viewing_source flag. I also removed it from the TabData class.